### PR TITLE
[FEAT]: Sql common table expressions (CTE's)

### PR DESCRIPTION
**Pull request by universalmind303**

### CODE REVIEW

File: src/daft-sql/src/planner.rs [hunk 1]
```diff
@@ -135,15 +200,24 @@ impl SQLPlanner {
...
+            SetExpr::Select(selection) => selection,
+            SetExpr::Query(_) => unsupported_sql_err!("Subqueries are not supported"),
+            SetExpr::SetOperation { .. } => {
+                unsupported_sql_err!("Set operations are not supported")
+            }
+            SetExpr::Values(..) => unsupported_sql_err!("VALUES are not supported"),
+            SetExpr::Insert(..) => unsupported_sql_err!("INSERT is not supported"),
+            SetExpr::Update(..) => unsupported_sql_err!("UPDATE is not supported"),
+            SetExpr::Table(..) => unsupported_sql_err!("TABLE is not supported"),
+        };
```
Comment (universalmind303) [inline src/daft-sql/src/planner.rs:213]: a small drive-by to provide better error messages here.

File: src/daft-sql/src/planner.rs [hunk 2]
```diff
@@ -497,12 +571,48 @@ impl SQLPlanner {
...
+                unsupported_sql_err!("Unsupported table factor: NestedJoin")
+            }
+            sqlparser::ast::TableFactor::Pivot { .. } => {
+                unsupported_sql_err!("Unsupported table factor: Pivot")
+            }
+            sqlparser::ast::TableFactor::Unpivot { .. } => {
+                unsupported_sql_err!("Unsupported table factor: Unpivot")
+            }
+            sqlparser::ast::TableFactor::MatchRecognize { .. } => {
+                unsupported_sql_err!("Unsupported table factor: MatchRecognize")
```
Comment (universalmind303) [inline src/daft-sql/src/planner.rs:615]: another small driveby for better error messages

**Review by kevinzwang — APPROVED** [review]
LGTM 🙌
